tools/functional-tester/etcd-runner/main.go

```diff
@@ -310,10 +320,24 @@ func runWatcher(eps []string) {
 						return
 					}
 				}
-			}(prefixes[i])
+			}(roundPrefix + "-" + prefixes[i])
 		}
 	}
 	wg.Wait()
+
+	// cancel all watch channels
+	cancelc()
+
```

Comment:
@xiang90, @heyitsanthony  this issue occurs only when a single context is being used for multiple watches. When each watch is assigned an independent context all channels get closed properly.